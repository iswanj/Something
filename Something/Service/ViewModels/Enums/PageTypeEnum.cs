namespace Service.ViewModels.Enums
{
    public enum PageTypeEnum
    {
        Login = 0,
        Register = 1,
        Onboarding = 2,
        Home = 3,
        Dashboard = 4,
    }
}
