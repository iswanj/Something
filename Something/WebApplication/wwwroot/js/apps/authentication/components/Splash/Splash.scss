.Splash {
    background: #ddd;
    color: #fff;
    background-image: url(/images/splash1.jpeg);
    background-position: center;
    height: 100%;
    &__login {
        background-position-x: 25%;
        background-position-y: 60%;
    }
    &__register {
        background-position-x: 90%;
        background-position-y: 60%;
    }
    &__inner {
        width: 400px;
        margin: 0 auto;
    }
}