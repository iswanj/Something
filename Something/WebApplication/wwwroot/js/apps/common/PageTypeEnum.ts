export enum PageTypes {
    LOGIN = 0,
    REGISTER = 1,
}
