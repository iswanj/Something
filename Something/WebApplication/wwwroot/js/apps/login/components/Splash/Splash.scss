.splash {
    background: linear-gradient(to bottom, #2f54eb 0%, #3a40d4 100%);
    color: #fff;
    &--inner {
        width: 400px;
        margin: 0 auto;
    }
    &--header {
        font-size: 32px;
    }
    img {
        margin-top: 30px;
        width: 400px;
    }
}