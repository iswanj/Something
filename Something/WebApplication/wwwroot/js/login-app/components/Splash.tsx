import * as React from 'react';

export class Splash extends React.Component {
    render() {
        return (
            <div className="pane splash ant-col-md-12">
                <div className="splash--inner">
                    <div className="splash--header">Explore Something!</div>
                    <img src="https://cdn.dribbble.com/users/250027/screenshots/4065967/gifmaker.org_gll2rd.gif" />
                </div>
            </div>
        );
    }
}