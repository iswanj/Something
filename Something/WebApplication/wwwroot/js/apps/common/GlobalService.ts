import { PageTypes } from "./PageTypeEnum";
import { IAuthenticationProps } from "../authentication/types/IAuthenticationProps";
import { BaseParams } from "./types/BaseParams";
import { IOnboardingProps } from "../onboarding/types/IOnboardingProps";

export const getParams = (pageType: PageTypes): BaseParams => {
    let params = null;

    switch (pageType) {
        case PageTypes.LOGIN:
        case PageTypes.REGISTER:
            {
                params = (window as any).authParams as IAuthenticationProps;
                break;
            }
        case PageTypes.ONBOARDING:
            {
                params = (window as any).onboardingParams as IOnboardingProps;
                break;
            }
        default:
            break;
    }
    return params;
} 