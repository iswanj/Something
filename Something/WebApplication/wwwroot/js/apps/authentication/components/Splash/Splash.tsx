import * as React from "react";
import "./Splash.scss";

export class Splash extends React.Component {
    render(): JSX.Element {
        return (
            <div className="Splash pane ant-col-md-12">
                <div className="Splash--inner">
                    <div className="Splash--header">Explore Something!</div>
                    <img src="https://cdn.dribbble.com/users/250027/screenshots/4065967/gifmaker.org_gll2rd.gif" />
                </div>
            </div>
        );
    }
}
