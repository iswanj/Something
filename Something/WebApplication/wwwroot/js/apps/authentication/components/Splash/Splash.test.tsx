import 'jest';
import * as React from 'react';
import { shallow } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';

import { Splash } from "./Splash";
import { PageTypes } from '../../../common/PageTypeEnum';

describe("<Splash />", () => {
    it("should render correctly", () => {
        const props = { pageType: PageTypes.LOGIN };
        const wrapper = shallow(<Splash {...props} />);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });
});