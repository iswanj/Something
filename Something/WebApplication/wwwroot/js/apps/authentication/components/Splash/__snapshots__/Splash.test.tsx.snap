// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Splash /> should render correctly 1`] = `
<div
  className="Splash Splash__login"
>
  <div
    className="Splash__inner"
  />
</div>
`;
